impeller/typographer/backends/skia/text_render_context_skia.h

```diff
@@ -6,9 +6,30 @@
 
 #include "flutter/fml/macros.h"
 #include "impeller/typographer/text_render_context.h"
+#include "third_party/skia/include/core/SkBitmap.h"
 
 namespace impeller {
 
+class DeviceBuffer;
+class Allocator;
+
+class FontImpellerAllocator : public SkBitmap::Allocator {
+ public:
+  explicit FontImpellerAllocator(
+      std::shared_ptr<impeller::Allocator> allocator);
+
+  ~FontImpellerAllocator() = default;
+
+  // |Allocator|
+  bool allocPixelRef(SkBitmap* bitmap) override;
+
+  std::optional<std::shared_ptr<DeviceBuffer>> GetDeviceBuffer() const;
+
+ private:
+  std::shared_ptr<impeller::Allocator> allocator_;
+  std::optional<std::shared_ptr<DeviceBuffer>> buffer_;
+};
+
```

Comment:
This is an exact copy of the implementation used in image decoding.

Reply:
I'd like to put it somewhere shared but I'm not quite sure where that would be.

Reply:
ignoring the skia dependency, it would just go in core. But with the skia dep I have no idea. I could create a new impeller-skia interop target?

Reply:
I don't think we should add a dependency on skia to core but we _could_. I think `impeller/image` is probably the right spot for this.